server/tserver/src/main/java/org/apache/accumulo/tserver/logger/LogReader.java

```diff
@@ -73,19 +77,37 @@
    * @param args
    *          - first argument is the file to print
    */
-  public static void main(String[] args) throws IOException {
+  public static void main(String[] args) throws Exception {
+    new LogReader().execute(args);
+  }
+
+  @Override
+  public String keyword() {
+    return "wal-info";
+  }
+
+  @Override
+  public String description() {
+    return "Prints WAL Info";
+  }
+
+  @SuppressFBWarnings(value = "DM_EXIT",
+      justification = "System.exit is fine here because it's a utility class executed by a main()")
+  @Override
+  public void execute(String[] args) throws Exception {
     Opts opts = new Opts();
-    opts.parseArgs(LogReader.class.getName(), args);
+    opts.parseArgs("accumulo wal-info", args);
+    if (opts.files.isEmpty()) {
+      System.err.println("No WAL files were given");
+      System.exit(1);
+    }
```

Comment:
This is okay, but I wonder if this will allow users to see the full usage. Is there a `--help` or `-?` option to show the usage, or if the command-line options are wrong, is there anyway for the user to see what the correct usage is? The code that this bit replaces did `new JCommander(opts).usage()`.

Reply:
There is a -h/--help and -? option available for the command. I followed the convention used with accumulo rfile-info. Initially I printed the given message and called the JCommander usage command. I wasn't particularly happy with what it printed out initially:
  Usage: <main class> [options] <logfile> { <logfile> ...}
I didn't find a way to have it replace '<main class>' with 'accumulo wal-info. The help options properly display 'accumulo wal-info ...' in the usage statement as desired. I'm okay putting the JCommander.usage back in but I wasn't sure if the <main class> verbiage would be confusing to users. Thoughts?

Reply:
I think this is consistent with what we do with the other KeywordExecutable utilities. Perhaps we could add an exit method to the interface to use by them all. The method could take the Help object since it has an exit method that's called by the JCommander parseArgs.

Reply:
The following code will print the message and then display the usage:

```
    if (opts.files.isEmpty()) { 
      JCommander jc = new JCommander(opts);
      jc.setProgramName("accumulo wal-info");
      System.err.println("No WAL files were given");
      jc.usage();
      System.exit(1);
    }
```
We can do this or merge the current changes and then place a follow-on ticket to implement Mike's suggestion.

Reply:
As long as there's some way for users to discover the usage, I don't necessarily care about the details. I just wasn't sure if the current changes still supported some kind of discovery. If you say there is a help flag already, then that's sufficient for me. If you want to make additional improvements later, that's also fine.

Reply:
I will merge with the current update and we can create a follow-on ticket if we decide to pursue additional improvements.